### PR TITLE
302 create mr volume element

### PR DESCRIPTION
**Pull request by michaelthatsit**
reimplemented `mr-volume` for clipping anchored content, even when it's not UI.

also made improvements to water shader and fixed an anchoring bug.

### CODE REVIEW

File: src/core/MREntity.js
Comment (hanbollar) [inline src/core/MREntity.js:69]: question - this should make all entities ignored by masking, even div entities - meaning we're not masking at all anymore - is this needed in this pr?

if so - make sure to add `this.ignoreStencil = false` to all mrdiventity items (or at a basis, mrdiventity)

**Review by hanbollar — APPROVED** [review]
one comment that needs looking at before merge - once that's resolved / lmk if you have questions - then lgtm ~

File: src/core/MREntity.js
Comment (michaelthatsit) [reply src/core/MREntity.js:69]: ah it should be disabled for MREntity but enabled for MRDivEntity, I'll fix that. This should achieve the desired effect of masking UI elements but allowing 3D content to be clipped in an mr-volume.

Comment (michaelthatsit) [reply src/core/MREntity.js:69]: @hanbollar looks like this is already the case in MRDivEntity, which sets it's values after the MREntities constructor runs.

https://github.com/Volumetrics-io/mrjs/blob/c802e57ef5f70f8f0ee3f25f915844c0550e745f/src/core/MRDivEntity.js#L44

Comment (hanbollar) [reply src/core/MREntity.js:69]: ah shoot - nah you're all good there, just checked and MRDivEntity already has the this.ignoreStencil = false ~, good to merge

**Pull request merged**
